x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -128,9 +129,8 @@ public Void run() {
     //private final Environment env;
     protected final Licensing licensing;
     // These should not be directly accessed as they cannot be overridden in tests. Please use the getters so they can be overridden.
-    private static final SetOnce<XPackLicenseState> licenseState = new SetOnce<>();
-    private static final SetOnce<SSLService> sslService = new SetOnce<>();
-    private static final SetOnce<LicenseService> licenseService = new SetOnce<>();
```

Comment:
The statics here relied on the existing order of calling `createComponents`. I decided to just fix this, since the new extension mechanism allowed us to delegate to a new `createComponents` method that includes the 3 services.